### PR TITLE
Ali

### PR DESCRIPTION
**Pull request by uprocker**
Hi @yehe01 
Can you do a review please?

### CODE REVIEW

File: app/src/main/java/com/example/blah/mobilestudio/Activities/MainActivity.java [hunk 1]
```diff
@@ -92,8 +94,10 @@ public void onFileSelected(String path) {
             }
         }
 
+        breadFragment = (BreadcrumbFragment) getFragmentManager().findFragmentById(R.id.topBreadFragment);
         if (breadFragment != null) {
             breadFragment.currentPath = path;
+            this.highlightBreadcrumbItem(path);
             breadFragment.onResume();
```
Comment (yehe01) [inline app/src/main/java/com/example/blah/mobilestudio/Activities/MainActivity.java]: Do we have to call life cycle methods explicitly?

File: app/src/main/java/com/example/blah/mobilestudio/breadcrumbview/BreadcrumbFragment.java [hunk 2]
```diff
@@ -35,20 +37,29 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         v = inflater.inflate(R.layout.breadcrumb_fragment, container, false);
         breadCrumb = (BreadcrumbView) v.findViewById(R.id.bread_bar);
+        if (savedInstanceState != null) {
+            currentPath = savedInstanceState.getString("currentPath");
+            breadCrumb.rootPath = savedInstanceState.getString("rootPath");
+            breadCrumb.highlightedItem = savedInstanceState.getString("highlightedItem");
+            breadCrumb.highlightedIndex= savedInstanceState.getInt("highlightedIndex");
```
Comment (yehe01) [inline app/src/main/java/com/example/blah/mobilestudio/breadcrumbview/BreadcrumbFragment.java]: Still, there is unformatted code. Can you format the whole file?

File: app/src/main/java/com/example/blah/mobilestudio/breadcrumbview/BreadcrumbFragment.java [hunk 3]
```diff
@@ -35,20 +37,29 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
+        }
         return v;
     }
 
     @Override
+    public void onStart() {
+        super.onStart();
+        breadCrumb = (BreadcrumbView) v.findViewById(R.id.bread_bar);
+        if (currentPath.length() > 0 && breadCrumb.rootPath .length() > 0) {
```
Comment (yehe01) [inline app/src/main/java/com/example/blah/mobilestudio/breadcrumbview/BreadcrumbFragment.java]: This one is not formatted. Interestingly, `irootPath. length()` can pass compling.

File: app/src/main/java/com/example/blah/mobilestudio/breadcrumbview/BreadcrumbFragment.java [hunk 4]
```diff
@@ -35,20 +37,29 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
...
         super.onResume();
         if (currentPath.length() > 0) {
-            // Remove the root folders path from the beginning of the selected path
-            currentPath = currentPath.replace(breadCrumb.rootPath, "");
-            // Retrive the sequence of the remaining folders in the path
-            ArrayList<String> items = new ArrayList(Arrays.asList(currentPath.substring(1).split("\\s*/\\s*")));
-            breadCrumb.SetElements(items);
-            if (breadCrumb.mListener != null)
-                breadCrumb.setOnClickListener(mListener);
+            CalculatePathAndSetTheListener(breadCrumb, currentPath);
```
Comment (yehe01) [inline app/src/main/java/com/example/blah/mobilestudio/breadcrumbview/BreadcrumbFragment.java]: `CalculatePathAndSetTheListener` shouldn't be capitalized as a convention.

File: app/src/main/java/com/example/blah/mobilestudio/breadcrumbview/BreadcrumbView.java [hunk 5]
```diff
@@ -178,7 +201,7 @@ public void onClick(View v) {
     }
 
     // This function allows the bacground image resizes based on the content size
-    public void SetBackgroundResizable(View tv) {
+    private void SetBackgroundResizable(View tv) {
```
Comment (yehe01) [inline app/src/main/java/com/example/blah/mobilestudio/breadcrumbview/BreadcrumbView.java]: Naming issue.

**Comment by uprocker** [comment]
formatting added, as well as some modifications.

**Pull request merged**